### ISSUE TITLE
Suggest an exact alternative in literal_coercion_linter

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1441]
is it always possible to suggest an exact alternative? I think it should be. so

as.integer(1) -> suggest 1L
as.numeric(NA) -> suggest NA_real_

etc. looking again I think it's definitely a big enough undertaking to warrant its own PR

_Originally posted by @MichaelChirico in https://github.com/r-lib/lintr/pull/1437#discussion_r912100833_

**Comment by MichaelChirico** [comment]
This is done already, AFAICT:

```r
lint(linters = literal_coercion_linter(), "
as.integer(NA)
dbl(NA)
as.numeric(1L)
")
```

Gives

```
<text>:2:1: warning: [literal_coercion_linter] Use NA_integer_ instead of as.integer(NA), i.e., use literals directly where possible, instead of coercion.
as.integer(NA)
^~~~~~~~~~~~~~
<text>:3:1: warning: [literal_coercion_linter] Use NA_real_ instead of dbl(NA), i.e., use literals directly where possible, instead of coercion.
dbl(NA)
^~~~~~~
<text>:4:1: warning: [literal_coercion_linter] Use 1 instead of as.numeric(1L), i.e., use literals directly where possible, instead of coercion.
as.numeric(1L)
^~~~~~~~~~~~~~
```
